tools/system_libs.py

```diff
@@ -1138,6 +1169,9 @@ def add_library(lib):
     # We need to build and link the library in
     add_library(lib)
 
+  while required_by_deps:
+    add_library(system_libs_map[required_by_deps.pop(0)])
```

Comment:
Is this to make dependencies show up at the end? I thought it doesn't really make a difference with `lld`.

In any case, if `required_by_deps` is supposed to be queue, may I suggest `collections.deque`?

Reply:
Reverted..